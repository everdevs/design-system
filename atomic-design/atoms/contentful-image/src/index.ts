export * from "./contentful-image";
export * from "./utils";
export * from "./types";
