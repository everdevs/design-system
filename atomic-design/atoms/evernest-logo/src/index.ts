export * from "./evernest-logo";
export * from "./types";
