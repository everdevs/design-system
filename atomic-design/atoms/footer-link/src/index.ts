export * from "./footer-link";
export * from "./types";
