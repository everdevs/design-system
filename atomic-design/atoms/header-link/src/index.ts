export * from "./header-link";
export * from "./types";
