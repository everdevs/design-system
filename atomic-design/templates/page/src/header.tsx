import styled from "@emotion/styled";
import { PropsWithTheme } from "@evernest/theme";

export const Header = styled.header<PropsWithTheme>`
	height: var(--header-height-ref);
`;
