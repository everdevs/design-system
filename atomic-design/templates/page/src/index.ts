export * from "./header";
export * from "./footer";
export * from "./main";
export * from "./page";
export * from "./styles";
