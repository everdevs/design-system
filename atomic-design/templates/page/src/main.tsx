import styled from "@emotion/styled";
import { PropsWithTheme } from "@evernest/theme";

export const Main = styled.main<PropsWithTheme>`
	display: flex;
	flex-direction: column;
	flex: 1;
	min-height: calc(100vh - var(--header-height-ref));
`;
