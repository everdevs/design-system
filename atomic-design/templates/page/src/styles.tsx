import { css, Global } from "@emotion/react";
import React from "react";
import { normalize } from "@evernest/normalize-css";

export const cssNormalize = css`
	${normalize};
`;

export const globalStyles = css`
	:root {
		--font-face-sans-serif: sans-serif;
		--font-face-serif: serif;
	}
	html {
		font-family: var(--font-face-sans-serif);
	}
	body {
		min-height: 100vh;
	}
	*,
	*::before,
	*::after {
		box-sizing: border-box;
	}
	a {
		color: currentColor;
	}
`;

export const Normalize = () => <Global styles={cssNormalize} />;

export const GlobalStyles = () => <Global styles={globalStyles} />;
