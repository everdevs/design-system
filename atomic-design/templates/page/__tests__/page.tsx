import React from "react";
import { mount } from "enzyme";
import { ThemedPageWithStyles } from "../src";
import { createTree } from "@evernest/dev-helpers";

test("ThemedPageWithStyles renders children", () => {
	const text = "Hello Page";
	const wrapper = mount(<ThemedPageWithStyles>{text}</ThemedPageWithStyles>);
	expect(wrapper.text()).toEqual(text);
});

test("ThemedPageWithStyles allows setting classname", () => {
	const className = "test";
	const wrapper = mount(<ThemedPageWithStyles className={className} />);
	expect(wrapper).toHaveClassName(className);
});

test("Page renders correctly", () => {
	const tree = createTree(<ThemedPageWithStyles />);
	expect(tree).toMatchSnapshot();
});
