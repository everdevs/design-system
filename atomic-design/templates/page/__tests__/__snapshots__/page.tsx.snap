// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page allows setting classname 1`] = `
<div
  className="css-153qske"
/>
`;
